services/supervisor/src/api.rs

```diff
@@ -59,6 +75,25 @@ impl PrivateApi for ApiImpl<'_> {
     fn start_service(&self, service: StartService) -> Result<Hash, Self::Error> {
         self.broadcast_transaction(service).map_err(From::from)
     }
+
+    fn confirm_config(&self, vote: ConfigVote) -> Result<Hash, Self::Error> {
+        self.broadcast_transaction(vote).map_err(From::from)
+    }
+}
+
+impl PublicApi for ApiImpl<'_> {
+    type Error = api::Error;
+
+    fn consensus_config(&self) -> Result<ConsensusConfig, Self::Error> {
+        Ok(CoreSchema::new(self.0.snapshot()).consensus_config())
+    }
+
+    fn config_proposal(&self) -> Result<ConfigProposalWithHash, Self::Error> {
+        Schema::new(self.0.instance.name, self.0.snapshot())
+            .pending_proposal()
+            .get()
+            .ok_or_else(|| Self::Error::NotFound("Pending config not found.".to_string()))
```

Comment:
I don't think that lack of pending proposal should be treated like error. Maybe `Ok(None)`?